hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/Invocation_OnMemberLeftTest.java

```diff
@@ -143,13 +179,32 @@ private void whenMemberRestarts(Runnable restartAction) throws Exception {
         }
     }
 
-    private void assertUnresponsiveOperationStarted() {
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                assertNotNull(remote.getUserContext().get(UnresponsiveTargetOperation.COMPLETION_FLAG));
-            }
-        });
+    private void assertUnresponsiveTargetOperationStarted() {
+        assertTrueEventually(() -> assertNotNull(remote.getUserContext().get(UnresponsiveTargetOperation.COMPLETION_FLAG)));
+    }
+
+    private void assertUnresponsiveMasterOperationStarted() {
+        assertTrueEventually(() -> assertNotNull(master.getUserContext().get(UnresponsiveMasterOperation.COMPLETION_FLAG)));
+    }
+
+    private static class UnresponsiveMasterOperation extends Operation {
+        static final String COMPLETION_FLAG = UnresponsiveMasterOperation.class.getName();
+
+        @Override
+        public void run() throws Exception {
+            getNodeEngine().getHazelcastInstance().getUserContext().put(COMPLETION_FLAG, new Object());
+        }
+
+        @Override
+        public boolean returnsResponse() {
+            return false;
+        }
+
+        @Override
+        public ExceptionAction onMasterInvocationException(Throwable throwable) {
+            MASTER_EXCEPTION_HANDLED.set(true);
+            return ExceptionAction.THROW_EXCEPTION;
```

Comment:
We should rather test that the operation is transparently retried with the new master after the old master is terminated.

Reply:
added